viite-backend/viite-main/src/main/scala/fi/liikennevirasto/viite/process/strategy/DefaultSectionCalculatorStrategy.scala

```diff
@@ -129,25 +129,58 @@ class DefaultSectionCalculatorStrategy extends RoadAddressSectionCalculatorStrat
   }
 
   private def continuousRoadwaySection(seq: Seq[ProjectLink], givenRoadwayNumber: Long): (Seq[ProjectLink], Seq[ProjectLink]) = {
-    val track =
-      seq.headOption.map(_.track).getOrElse(Track.Unknown)
-    val discontinuity =
-      seq.head.discontinuity
-    val administrativeClass =
-      seq.head.administrativeClass
-    val continuousProjectLinks =
-      seq.takeWhile(pl => pl.track == track && pl.discontinuity == discontinuity && pl.administrativeClass == administrativeClass)
+    def pickGeometricallyConnectedSection(startLink: ProjectLink, allLinks: Seq[ProjectLink]): List[ProjectLink] = {
+      def getSection(currentLink: ProjectLink, section: List[ProjectLink]): List[ProjectLink] = {
+        val currentIndex = allLinks.indexOf(currentLink)
+        val nextIndex = currentIndex + 1
+
+        if (nextIndex < allLinks.length) {
+          val nextLink = allLinks(nextIndex)
+
+          if (GeometryUtils.areGeometriesConnected(currentLink.startingPoint, currentLink.endPoint, nextLink.startingPoint, nextLink.endPoint)) {
+            getSection(nextLink, section :+ currentLink)
+          } else {
+            section :+ currentLink
+          }
+        } else {
+          section :+ currentLink
+        }
+      }
+
+      getSection(startLink, List.empty[ProjectLink])
+    }
+
+    val track = seq.headOption.map(_.track).getOrElse(Track.Unknown)
```

Comment:
Hmm nii, sanoit et mietit jos ottais ton seq.head:in muuttujaan, pingaan varmuuden vuoks :)

Reply:
muuttujan nimi seq on surkea.

Reply:
Joo tuo on tarkoitus nimetä uudelleen